lib/updateDeps.js

```diff
@@ -0,0 +1,166 @@
+
+	// Release type found by commitAnalyzer.
+	if (pkg._nextType) {
+		return pkg._nextType;
+	}
+
+	if (!dependentReleaseType) {
+		return undefined;
+	}
+
+	pkg._nextType = releaseStrategy === "inherit" ? dependentReleaseType : releaseStrategy;
+
+	return pkg._nextType;
+};
+
+/**
+ * Get dependent release type by recursive scanning and updating its deps.
+ *
+ * @param {Package} pkg The package with local deps to check.
+ * @param {string} bumpStrategy Dependency resolution strategy: override, satisfy, inherit.
+ * @param {string} releaseStrategy Release type triggered by deps updating: patch, minor, major, inherit.
+ * @param {Package[]} ignore Packages to ignore (to prevent infinite loops).
+ * @returns {string|undefined} Returns the highest release type if found, undefined otherwise
+ * @internal
+ */
+const getDependentRelease = (pkg, bumpStrategy, releaseStrategy, ignore) => {
+	const severityOrder = ["patch", "minor", "major"];
+	const { localDeps, manifest = {} } = pkg;
+	const { dependencies = {}, devDependencies = {}, peerDependencies = {}, optionalDependencies = {} } = manifest;
+	const scopes = [dependencies, devDependencies, peerDependencies, optionalDependencies];
+	const bumpDependency = (scope, name, nextVersion) => {
+		const currentVersion = scope[name];
+		if (!nextVersion || !currentVersion) {
+			return;
+		}
+		const resolvedVersion = resolveNextVersion(currentVersion, nextVersion, releaseStrategy);
+
+		if (currentVersion !== resolvedVersion) {
+			scope[name] = resolvedVersion;
+
+			return true;
+		}
+	};
+
+	// prettier-ignore
+	return localDeps
+		.filter((p) => ignore.indexOf(p) === -1)
+		.reduce((releaseType, p) => {
+			const name = p.name;
+
+			// Has changed if...
+			// 1. Any local dep package itself has changed
+			// 2. Any local dep package has local deps that have changed.
+			const nextType = resolveReleaseType(p, bumpStrategy, releaseStrategy,[...ignore, ...localDeps]);
+			const nextVersion = getNextVersion(p);
+			const lastVersion = pkg._lastRelease && pkg._lastRelease.version;
+
+			// 3. And this change should correspond to manifest updating rule.
+			const requireRelease = [
+				...scopes.map((scope) => bumpDependency(scope, name, nextVersion)),
+			].some(v => v) || !lastVersion;
+
+			return requireRelease && (severityOrder.indexOf(nextType) > severityOrder.indexOf(releaseType))
+				? nextType
+				: releaseType;
+		}, undefined);
+};
+
+/**
+ * Resolve next version of dependency.
+ *
+ * @param {string} currentVersion Current dep version
+ * @param {string} nextVersion Next release type: patch, minor, major
+ * @param {string|undefined} strategy Resolution strategy: inherit, override, satisfy
+ * @returns {string} Next dependency version
+ * @internal
+ */
+const resolveNextVersion = (currentVersion, nextVersion, strategy = "override") => {
+	if (strategy === "satisfy" && semver.satisfies(nextVersion, currentVersion)) {
+		return currentVersion;
+	}
+
+	if (strategy === "inherit") {
```

Comment:
Can you document what the intended outcome of this strategy is? I'm not sure I fully understand what it's meant to do?

Reply:
It looks like it could just be a simple assignment; not sure why you need the maps & replace in there, given how semver specifies a version?

Reply:
Copypasted some comments from README.md. https://github.com/dhoulb/multi-semantic-release/pull/45/commits/ea190073f1a6a1176ec6aa5273b06896aa9e31b3

It looks possible to create `semver.Range` instance, modify its comparators and format string back. But, I'm afraid, this can take too much time.

```js
const range = new semver.Range("^1.0.0");
Range {
  options: { loose: false, includePrerelease: false },
  loose: false,
  includePrerelease: false,
  raw: '^1.0.0',
  set: [ [ [Comparator], [Comparator] ] ],
  range: '>=1.0.0 <2.0.0-0'
}
> range.set[0]
[
  Comparator {
    options: { loose: false, includePrerelease: false },
    loose: false,
    operator: '>=',
    semver: SemVer {
      options: [Object],
      loose: false,
      includePrerelease: false,
      raw: '1.0.0',
      major: 1,
      minor: 0,
      patch: 0,
      prerelease: [],
      build: [],
      version: '1.0.0'
    },
    value: '>=1.0.0'
  },
  Comparator {
    options: { loose: false, includePrerelease: false },
    loose: false,
    operator: '<',
    semver: SemVer {
      options: [Object],
      loose: false,
      includePrerelease: false,
      raw: '2.0.0-0',
      major: 2,
      minor: 0,
      patch: 0,
      prerelease: [Array],
      build: [],
      version: '2.0.0-0'
    },
    value: '<2.0.0-0'
  }
]
```

Reply:
What about doing a `semver.inc` on the value, and then check if the original value satisfies that new value?